diff --git a/python/triton/_internal_testing.py b/python/triton/_internal_testing.py
index 27195d555f..b96313451a 100644
--- a/python/triton/_internal_testing.py
+++ b/python/triton/_internal_testing.py
@@ -42,6 +42,9 @@ def is_cuda():
     target = get_current_target()
     return False if target is None else target.backend == "cuda"
 
+def is_cpu():
+    target = get_current_target()
+    return False if target is None else target.backend == "cpu"
 
 def is_ampere_or_newer():
     return is_cuda() and torch.cuda.get_device_capability()[0] >= 8
