diff --git a/lib/Conversion/TritonGPUToLLVM/ControlFlowOpToLLVM.cpp b/lib/Conversion/TritonGPUToLLVM/ControlFlowOpToLLVM.cpp
index f7fd90b8bd..67e45ab165 100644
--- a/lib/Conversion/TritonGPUToLLVM/ControlFlowOpToLLVM.cpp
+++ b/lib/Conversion/TritonGPUToLLVM/ControlFlowOpToLLVM.cpp
@@ -61,7 +61,10 @@ struct CallOpConversion : public ConvertOpToLLVMPattern<triton::CallOp> {
   matchAndRewrite(triton::CallOp callOp,
                   typename triton::CallOp::Adaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    auto promotedOperands = promoteOperands(callOp, adaptor, rewriter);
+    // auto promotedOperands = promoteOperands(callOp, adaptor, rewriter);
+     auto promotedOperands = this->getTypeConverter()->promoteOperands(
+        callOp.getLoc(), /*opOperands=*/callOp->getOperands(),
+        adaptor.getOperands(), rewriter);
     auto newCallOp =
         convertCallOpToLLVMCallOp(callOp, promotedOperands, rewriter);
     if (!newCallOp)
