diff --git a/lib/Dialect/Triton/Transforms/RewriteTensorDescriptorToPointer.cpp b/lib/Dialect/Triton/Transforms/RewriteTensorDescriptorToPointer.cpp
index 4076a460c..afd6f9579 100644
--- a/lib/Dialect/Triton/Transforms/RewriteTensorDescriptorToPointer.cpp
+++ b/lib/Dialect/Triton/Transforms/RewriteTensorDescriptorToPointer.cpp
@@ -605,7 +605,6 @@ class TritonRewriteTensorDescriptorToPointerPass
 
     ConversionConfig config;
     config.buildMaterializations = false;
-    config.allowPatternRollback = false;
 
     if (mlir::failed(mlir::applyPartialConversion(
             op, target, std::move(patterns), config))) {
