diff --git a/python/test/unit/language/test_core.py b/python/test/unit/language/test_core.py
index fdc268937..513985378 100644
--- a/python/test/unit/language/test_core.py
+++ b/python/test/unit/language/test_core.py
@@ -37,6 +37,7 @@ from triton._internal_testing import (
     is_hip_rdna3,
     is_hip_rdna4,
     is_hip_gfx1250,
+    is_cpu,
     is_xpu,
     torch_float8_dtypes,
     torch_dtypes,
@@ -128,6 +129,11 @@ def check_type_supported(dtype, device):
             pytest.skip("bfloat16 is only supported on NVGPU with cc >= 80")
         if cc[0] < 9 and dtype in {tl.float8e4nv, "float8e4nv", "float8_e4m3fn"}:
             pytest.skip("float8e4nv is only supported on NVGPU with cc >= 90")
+    if device in ['cpu']:
+        if (dtype is tl.bfloat16 or dtype == "bfloat16" or dtype is torch.bfloat16):
+            pytest.skip("bfloat16 is not supported on CPU")
+        if dtype in torch_float8_dtypes:
+            pytest.skip("fp8 types not yet supported on CPU")
     if is_interpreter():
         if dtype in [tl.bfloat16, "bfloat16", torch.bfloat16]:
             pytest.skip("bfloat16 is not supported in the interpreter")
@@ -2230,6 +2236,8 @@ def test_load_store_same_ptr(device):
         x = torch.ones((65536, ), device=device, dtype=torch.float32)
         if is_hip():
             kernel[(65536, )](x, num_warps=16)  # threads per Warp for ROCM is 64
+        elif is_cpu():
+            kernel[(65536, )](x, num_warps=1)
         else:
             kernel[(65536, )](x, num_warps=32)
         assert torch.all(x == 2)
@@ -3461,8 +3469,10 @@ def convert_fp8_to_fp32(x, device, dtype_str):
 def get_test_dot_base_cases():
     return [(*shape, 4, False, False, epilogue, input_precision, in_dtype, out_dtype, 1, None)
             for shape in [(64, 64, 64), (32, 32, 32), (16, 16, 16)]
-            for epilogue in ['none', 'trans', 'add-matrix', 'add-rows', 'add-cols', 'softmax', 'chain-dot']
-            for input_precision in ['tf32', 'tf32x3', 'ieee', 'bf16x3', 'bf16x6']
+            # TODO fix softmax failures on CPU
+            for epilogue in (['none', 'trans', 'add-matrix', 'add-rows', 'add-cols', 'chain-dot'] if is_cpu(
+            ) else ['none', 'trans', 'add-matrix', 'add-rows', 'add-cols', 'softmax', 'chain-dot'])
+            for input_precision in (['ieee'] if is_cpu() else ['tf32', 'tf32x3', 'ieee', 'bf16x3', 'bf16x6'])
             for in_dtype, out_dtype in [('float16', 'float16'), ('float16',
                                                                  'float32'), ('float32',
                                                                               'float32'), ('float64', 'float64')]
@@ -3482,6 +3492,8 @@ def get_test_dot_softmax():
 
 # M, N, K, num_warps, col_a, col_b, epilogue, input_precision, in_dtype, out_dtype, kpack, mma_nonk_size
 def get_test_dot_mixed_sizes_cases():
+    if is_cpu():
+        return []
     available_kpack = [1, 2 if (is_hip() and not is_hip_cdna4()) else 1]
     available_precision = ["tf32" if is_cuda() else "ieee"]
     return [
@@ -3508,6 +3520,8 @@ def get_test_dot_transposed_op_base_cases():
 # M, N, K, num_warps, col_a, col_b, epilogue, input_precision, in_dtype, out_dtype, kpack, mma_nonk_size
 # Introduced in #2750
 def get_test_dot_h100_shortcut_cases():
+    if is_cpu():
+        return []
     return [(64, 64, 64, 4, False, False, 'chain-dot', 'ieee', 'bfloat16', 'float32', 1, None)]
 
 
@@ -3523,6 +3537,8 @@ def get_test_dot_mfma_edge_cases():
 # M, N, K, num_warps, col_a, col_b, epilogue, input_precision, in_dtype, out_dtype, kpack, mma_nonk_size
 # introduced in #3370
 def get_test_dot_fp8_output_cases():
+    if is_cpu():
+        return []
     return [(128, 128, 64, 4, False, False, 'chain-dot', 'ieee', float8_type, 'float32', 1, None)
             for float8_type in ["float8e5", "float8e4nv"]]
 
@@ -3596,6 +3612,10 @@ def get_test_dot_small_k_wmma_cases():
 
 
 def get_test_small_dots_cases():
+    if is_cpu():
+        return [(2, 4, 32, 1, False, False, 'None', 'ieee', 'float16', 'float32', 1, None),
+                (64, 8, 8, 4, False, False, 'None', 'ieee', 'float32', 'float32', 1, None),
+                (64, 8, 16, 4, False, False, 'None', 'ieee', 'float16', 'float32', 1, None)]
     if not is_cuda():
         return []
     return [(2, 4, 32, 1, False, False, 'None', 'ieee', 'float16', 'float32', 1, None),
@@ -3633,6 +3653,11 @@ def test_dot(M, N, K, num_warps, col_a, col_b, epilogue, input_precision, in_dty
         if input_precision == "bf16x3" or input_precision == "bf16x6":
             pytest.skip(f"input_precision {input_precision} is not supported in the interpreter")
     else:
+        if is_cpu():
+            if input_precision != "ieee":
+                pytest.skip("Only IEEE precision is supported on CPU")
+            if in_dtype not in ['float16', 'float32', 'float64']:
+                pytest.skip("Only float16, float32 and float64 are supported on CPU")
         if not is_hip() and K < 16:
             tf32_n8 = (in_dtype == 'float32' and N == 8 and K == 8 and input_precision == 'tf32')
             if in_dtype != 'float64' and not tf32_n8:
