diff --git a/include/triton/Conversion/TritonGPUToLLVM/Utility.h b/include/triton/Conversion/TritonGPUToLLVM/Utility.h
index bb800f4ee..21d643a6b 100644
--- a/include/triton/Conversion/TritonGPUToLLVM/Utility.h
+++ b/include/triton/Conversion/TritonGPUToLLVM/Utility.h
@@ -642,7 +642,7 @@ SmallVector<Value> inlineRegion(RewriterBase &rewriter, Region &region,
 // }
 // #thenBlock
 std::tuple</*prevBlock=*/Block *, /*ifBlock=*/Block *, /*thenBlock=*/Block *>
-createIfBlock(ConversionPatternRewriter &b, Location loc, Value cnd);
+createIfBlock(RewriterBase &b, Location loc, Value cnd);
 
 void finalizeTensorAtomicResults(Operation *op, RankedTensorType tensorTy,
                                  ConversionPatternRewriter &rewriter,
